lib/internal/streams/pipeline.js

```diff
@@ -147,6 +153,63 @@ async function pump(iterable, writable, finish, { end }) {
   }
 }
 
+async function pumpToWeb(readable, writable, finish, { end }) {
+  let error;
+  let onresolve = null;
+
+  if (isTransformStream(writable)) {
+    writable = writable.writable;
+  }
+
+  const resume = (err) => {
+    if (err) {
+      error = err;
+    }
+
+    if (onresolve) {
+      const callback = onresolve;
+      onresolve = null;
+      callback();
+    }
+  };
+
+  const wait = () => new Promise((resolve, reject) => {
+    if (error) {
+      reject(error);
+    } else {
+      onresolve = () => {
+        if (error) {
+          reject(error);
+        } else {
+          resolve();
+        }
+      };
+    }
+  });
+
+  eos(writable, resume);
+
+  try {
+    const writer = writable.getWriter();
+    for await (const chunk of readable) {
+      const written = writer.write(chunk);
+      if (!written) {
+        await wait();
```

Comment:
> [Return value](https://developer.mozilla.org/en-US/docs/Web/API/WritableStreamDefaultWriter/write#return_value)
A [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise), which fulfills with the undefined upon a successful write, or rejects if the write fails or stream becomes errored before the writing process is initiated.

```suggestion
      const written = await writer.write(chunk);
      if (!written) {
       // TODO: What does this mean?
```